File: java/fastcampus-java-spring/part2/Chapter6/src/ch20/ThreadTest.java
package ch20;

class MyThread implements Runnable {
	public void run() {
		int i;
		
		for (i = 0; i < 200; i++) {
			System.out.print(i + "\t");
		}
		
	}
}

public class ThreadTest {

	public static void main(String[] args) {
		System.out.println(Thread.currentThread() + "start");
		MyThread runnable = new MyThread();
		
		Thread th1 = new Thread(runnable);
		Thread th2 = new Thread(runnable);
		
		th1.start();
		th2.start();

		System.out.println(Thread.currentThread() + "end");
		
		Runnable run = new Runnable() {

			@Override
			public void run() {
				System.out.println("run");
			}
			
		};
		
		run.run();
	}

}

File: kotlin/boostcourse-kotlin-basic-2/src/chap06/section1/ThreadTest.kt
package chap06.section1

class SimpleThread: Thread() {
    override fun run() {
        println("Class Thread: ${Thread.currentThread()}")
    }
}

class SimpleRunnable: Runnable {
    override fun run() {
        println("Interface Thread: ${Thread.currentThread()}")
    }
}

fun main() {
    val thread = SimpleThread()
    thread.start()

    val runnable = SimpleRunnable()
    val thread2 = Thread(runnable)
    thread2.start()

    // 익명객체
    object: Thread() {
        override fun run() {
            println("object Thread: ${Thread.currentThread()}")
        }
    }.start()

    // 람다식
    Thread {
        println("Lambda Thread: ${Thread.currentThread()}")
    }.start()
}